src/pybind/mgr/cephadm/module.py

```diff
@@ -3985,6 +4000,47 @@ def remove_osds_status(self) -> List[OSD]:
         """
         return self.to_remove_osds.all_osds()
 
+    @handle_orch_error
+    @service_exists()
+    def set_osd_spec(self, service_name: str, osd_ids: List[str]) -> str:
+        """
+        Update unit.meta file for osd with service name
+        """
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_type('osd')
+        update_osd = defaultdict(list)
+        for daemon in daemons:
+            if daemon.daemon_id in osd_ids:
+                if not daemon.hostname or not daemon.daemon_name:
+                    continue
+                update_osd[daemon.hostname].append(daemon.daemon_name)
+
+        if not update_osd:
+            raise OrchestratorError(f"Unable to find OSDs: {osd_ids}")
+
+        failed_osd = []
+        success_osd = []
+        for host in update_osd:
+            osds = ",".join(update_osd[host])
+            # run cephadm command with all host osds on specific host,
+            # if it fails, continue with other hosts
+            try:
+                with self.async_timeout_handler(host):
+                    outs, errs, _code = self.wait_async(
+                        CephadmServe(self)._run_cephadm(host,
+                                                        cephadmNoImage,
+                                                        'update-osd-service',
+                                                        ['--service_name', service_name, '--name', osds]))
+                    if _code:
+                        self.log.error(f"Failed to update service for {osds} osd. Cephadm error: {errs}")
+                        failed_osd.extend(update_osd[host])
+                    else:
+                        success_osd.extend(update_osd[host])
+            except Exception:
+                self.log.exception("Failed to set service name for {osds}")
+                failed_osd.extend(update_osd[host])
+            self._invalidate_daemons_and_kick_serve(host)
+        return f"Updated service for osd {success_osd} and failed for {failed_osd}"
```

Comment:
NIT
```suggestion
        return f"Updated service for osd {success_osd} {'and failed for ' + ','.join(failed_osd) if failed_osd else ''}"
```
or something sort of like that (I didn't test what I wrote above so I don't know if it actually works). Just so we only add the `and failed for` if we're going to actually report failures